🐛 [firebase_core] `flutter build windows --profile` fails with `error LNK2001: unresolved external symbol __imp__invalid_parameter`

## Bug report

**Describe the bug**

`flutter build windows --profile` fails with `error LNK2001: unresolved external symbol __imp__invalid_parameter` errors:
```
C:\dev\Flutter\firebase_windows_profile_bug>flutter build windows --profile

firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_filesystem_desktop_windows.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_reference_counted_future_impl.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_scheduler.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_base64.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_callback.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_cleanup_notifier.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]       
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_common.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]     
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]  
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_filesystem_desktop_windows.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]  
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_reference_counted_future_impl.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]   
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_scheduler.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_base64.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_callback.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_cleanup_notifier.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_common.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static void * __cdecl std::_Crt_new_delete::operator new(unsigned __int64)" (__imp_??2_Crt_new_delete@std@@SAPEAX_K@Z) referenced in function "public: __cdecl std::wstring_convert<class std::codecvt_utf8_utf16<wchar_t,1114111,0>,wchar_t,class stReportMode referenced in function "void __cdecl f_b_flatbuffers::SetupDefaultCRTReportMode(void)" (?SetupDefaultCRTReportMode@f_b_flatbuffers@@YAXXZ) [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]
C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\Profile\firebase_windows_profile_bug.exe : fatal error LNK1120: 7 unresolved externals [C:\dev\Flutter\firebase_windows_profile_bug\build\windows\x64\runner\firebase_windows_profile_bug.vcxproj]Building Windows application...                                   315.6s
Build process failed.
```


### Steps to reproduce

Steps to reproduce the behavior:

1. Add `firebase_core` dependency to any project.
2. Run `flutter build windows --profile`.
3. See the error.

### Expected behavior

Application builds.

### Sample project

https://github.com/SleepySquash/firebase_windows_profile_bug

---

## Additional context

#11997 fixes the issue for `--release` build, yet `--profile` still fails. Yes, `flutter build windows --release` does work.

Temporary workaround is to specify this in the `windows/CMakeLists.txt`:
```cmake
set(CMAKE_BUILD_TYPE "Release")
```

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click to Expand</summary>

```
[√] Flutter (Channel stable, 3.16.4, on Microsoft Windows [Version 10.0.22621.2861], locale en-US)
    • Flutter version 3.16.4 on channel stable at C:\dev\env\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2e9cb0aa71 (7 days ago), 2023-12-11 14:35:13 -0700
    • Engine revision 54a7145303
    • Dart version 3.2.3
    • DevTools version 2.28.4

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\dev\env\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.6.4)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.6.33815.320
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code, 64-bit edition (version 1.71.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.2861]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 120.0.6099.109
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 117.0.2045.60

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 3.2.3
Flutter SDK 3.16.4
firebase_windows_profile_bug 1.0.0+1

dependencies:
- cupertino_icons 1.0.6
- firebase_core 2.24.2 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math web sky_engine]

dev dependencies:
- flutter_lints 2.0.3 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math async boolean_selector characters collection material_color_utilities meta source_span stream_channel string_scanner term_glyph web]

transitive dependencies:
- async 2.11.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.3.0
- clock 1.1.1
- collection 1.18.0
- fake_async 1.3.1 [clock collection]
- firebase_core_platform_interface 5.0.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.10.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math web]
- js 0.6.7 [meta]
- lints 2.1.1
- matcher 0.12.16 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.5.0 [collection]
- meta 1.10.0
- path 1.8.3
- plugin_platform_interface 2.1.7 [meta]
- sky_engine 0.0.99
- source_span 1.10.0 [collection path term_glyph]
- stack_trace 1.11.1 [path]
- stream_channel 2.1.2 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.6.1 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- vector_math 2.1.4
- web 0.3.0
```

</details>

---

I am experiencing the same issue. Here are the logs:

<details>

<summary> Logs </summary>

flutter clean; flutter pub get; flutter run -d windows --profile       
Deleting build...                                                   3,7s
Deleting .dart_tool...                                             184ms
Deleting Generated.xcconfig...                                       2ms
Deleting build...                                                   3,7s
Deleting .dart_tool...                                             184ms
Deleting .dart_tool...                                             184ms
Deleting Generated.xcconfig...                                       2ms
Deleting flutter_export_environment.sh...                            1ms
Deleting ephemeral...                                            1.730ms
Deleting ephemeral...                                                1ms
Deleting ephemeral...                                              350ms
Deleting .flutter-plugins-dependencies...                           13ms
Deleting .flutter-plugins...                                        10ms
Resolving dependencies... (3.9s)
Downloading packages... (4.5s)
  async 2.11.0 (2.12.0 available)
  boolean_selector 2.1.1 (2.1.2 available)
  change_case 1.1.0 (2.1.0 available)
  characters 1.3.0 (1.4.0 available)
  clock 1.1.1 (1.1.2 available)
  collection 1.18.0 (1.19.1 available)
  fake_async 1.3.1 (1.3.2 available)
  file_picker 8.1.6 (8.1.7 available)
  flutter_secure_storage 8.1.0 (9.2.2 available)
  flutter_secure_storage_windows 2.1.1 (3.1.2 available)
  flutter_svg 1.1.6 (2.0.16 available)
  fvp 0.24.1 (0.29.0 available)
  get_it 7.7.0 (8.0.3 available)
  http_parser 4.0.2 (4.1.1 available)
! intl 0.18.1 (overridden) (0.20.1 available)
  js 0.6.7 (0.7.1 available)
  leak_tracker 10.0.5 (10.0.8 available)
  leak_tracker_flutter_testing 3.0.5 (3.0.9 available)
  lints 5.0.0 (5.1.1 available)
  matcher 0.12.16+1 (0.12.17 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  meta 1.15.0 (1.16.0 available)
  path 1.9.0 (1.9.1 available)
  safe_local_storage 1.0.2 (2.0.1 available)
  screen_brightness 0.2.2+1 (2.0.1 available)
  screen_brightness_android 0.1.0+2 (2.0.0 available)
  screen_brightness_ios 0.1.0 (2.0.0 available)
  screen_brightness_macos 0.1.0+1 (2.0.0 available)
  screen_brightness_platform_interface 0.1.0 (2.0.0 available)
  screen_brightness_windows 0.1.3 (2.0.1 available)
  source_span 1.10.0 (1.10.1 available)
  stack_trace 1.11.1 (1.12.1 available)
  stream_channel 2.1.2 (2.1.3 available)
  string_scanner 1.2.0 (1.4.1 available)
  term_glyph 1.2.1 (1.2.2 available)
  test_api 0.7.2 (0.7.4 available)
  uri_parser 2.0.2 (3.0.0 available)
  vm_service 14.2.5 (15.0.0 available)
  win32_registry 1.1.5 (2.0.1 available)
Got dependencies!
39 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Launching lib\main.dart on Windows in profile mode...
CMake Deprecation Warning at unity/build/windows/x64/extracted/firebase_cpp_sdk_windows/CMakeLists.txt:17 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_filesystem_desktop_windows.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_reference_counted_future_impl.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_util.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_scheduler.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_base64.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_callback.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_cleanup_notifier.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_common.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: símbolo externo não resolvido __imp__invalid_parameter [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]    
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: símbolo externo não resolvido __imp__calloc_dbg [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_filesystem_desktop_windows.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport [unity\build\windows\x64\runner\bluecherry_client.vcxproj]  
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_reference_counted_future_impl.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_util.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport [unity\build\windows\x64\runner\bluecherry_client.vcxproj] 
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport [unity\build\windows\x64\runner\bluecherry_client.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_scheduler.obj) : error LNK2001: símbolo externo não resolvido __imp__CrtDbgReport 
Building Windows application...                                   585,1s
Error: Build process failed.
#0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
#1      _runBuild (package:flutter_tools/src/windows/build_windows.dart:266:5)
<asynchronous suspension>
#2      buildWindows (package:flutter_tools/src/windows/build_windows.dart:111:5)
<asynchronous suspension>
#3      WindowsDevice.buildForDevice (package:flutter_tools/src/windows/windows_device.dart:65:5)
<asynchronous suspension>
#4      DesktopDevice.startApp (package:flutter_tools/src/desktop_device.dart:121:7)
<asynchronous suspension>
#5      FlutterDevice.runCold (package:flutter_tools/src/resident_runner.dart:534:33)
<asynchronous suspension>
#6      ColdRunner.run (package:flutter_tools/src/run_cold.dart:55:28)
<asynchronous suspension>
#7      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:867:27)
<asynchronous suspension>
#8      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1408:27)
<asynchronous suspension>
#9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#10     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#11     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:420:9)
<asynchronous suspension>
#12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#13     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:364:5)
<asynchronous suspension>
#14     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:130:9)
<asynchronous suspension>
#15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#16     main (package:flutter_tools/executable.dart:93:3)
<asynchronous suspension>

</details>

The error does not happen in debug or release mode.

Facing the same issue while running in --profile mode on windows. Below attached the logs for reference.

<details>
<summary>Logs</summary>

```dart

firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_filesystem_desktop_windows.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_reference_counted_future_impl.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_scheduler.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_base64.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_callback.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_cleanup_notifier.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_common.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: unresolved external symbol __imp__invalid_parameter [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: unresolved external symbol __imp__calloc_dbg [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_filesystem_desktop_windows.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_variant_util.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_reference_counted_future_impl.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_util.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_scheduler.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_base64.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_controller_desktop.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_idl_parser.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_callback.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_cleanup_notifier.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_options.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_common.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_date_provider.obj) : error LNK2001: unresolved external symbol __imp__CrtDbgReport [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static void * __cdecl std::_Crt_new_delete::operator new(unsigned __int64)" (__imp_??2_Crt_new_delete@std@@SAPEAX_K@Z) referenced in function "public: __cdecl std::wstring_convert<class std::codecvt_utf8_utf16<wchar_t,1114111,0>,wchar_t,class std::allocator<wchar_t>,class std::allocator<char> >::wstring_convert<class std::codecvt_utf8_utf16<wchar_t,1114111,0>,wchar_t,class std::allocator<wchar_t>,class std::allocator<char> >(void)" (??0?$wstring_convert@V?$codecvt_utf8_utf16@_W$0BAPPPP@$0A@@std@@_WV?$allocator@_W@2@V?$allocator@D@2@@std@@QEAA@XZ) [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol "__declspec(dllimport) public: static void * __cdecl std::_Crt_new_delete::operator new(unsigned __int64)" (__imp_??2_Crt_new_delete@std@@SAPEAX_K@Z) [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_app_desktop.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static void __cdecl std::_Crt_new_delete::operator delete(void *)" (__imp_??3_Crt_new_delete@std@@SAXPEAX@Z) referenced in function "public: virtual void * __cdecl std::codecvt_utf8_utf16<wchar_t,1114111,0>::`scalar deleting destructor'(unsigned int)" (??_G?$codecvt_utf8_utf16@_W$0BAPPPP@$0A@@std@@UEAAPEAXI@Z) [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d94e1bff3c0e8414a6679ea91c3e103e_firebase_app.dir_Debug_heartbeat_storage_desktop.obj) : error LNK2001: unresolved external symbol "__declspec(dllimport) public: static void __cdecl std::_Crt_new_delete::operator delete(void *)" (__imp_??3_Crt_new_delete@std@@SAXPEAX@Z) [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2019: unresolved external symbol __imp__CrtSetReportFile referenced in function "void __cdecl f_b_flatbuffers::SetupDefaultCRTReportMode(void)" (?SetupDefaultCRTReportMode@f_b_flatbuffers@@YAXXZ) [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
firebase_app.lib(d3d7d08a438878e74aeb2cbaaedfc967_flatbuffers.dir_Debug_util.obj) : error LNK2019: unresolved external symbol __imp__CrtSetReportMode referenced in function "void __cdecl f_b_flatbuffers::SetupDefaultCRTReportMode(void)" (?SetupDefaultCRTReportMode@f_b_flatbuffers@@YAXXZ) [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
D:\project_dir\build\windows\x64\runner\Profile\smart_billing.exe : fatal error LNK1120: 7 unresolved externals [D:\project_dir\build\windows\x64\runner\smart_billing.vcxproj]
Error: Build process failed.
#0      throwToolExit (package:flutter_tools/src/base/common.dart:34:3)
#1      _runBuild (package:flutter_tools/src/windows/build_windows.dart:259:5)
<asynchronous suspension>
#2      buildWindows (package:flutter_tools/src/windows/build_windows.dart:114:5)
<asynchronous suspension>
#3      WindowsDevice.buildForDevice (package:flutter_tools/src/windows/windows_device.dart:65:5)
<asynchronous suspension>
#4      DesktopDevice.startApp (package:flutter_tools/src/desktop_device.dart:108:7)
<asynchronous suspension>
#5      FlutterDevice.runCold (package:flutter_tools/src/resident_runner.dart:524:33)
<asynchronous suspension>
#6      ColdRunner.run (package:flutter_tools/src/run_cold.dart:52:28)
<asynchronous suspension>
#7      AppDomain.launch.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:823:9)
<asynchronous suspension>
#8      AppDomain.launch (package:flutter_tools/src/commands/daemon.dart:821:5)
<asynchronous suspension>
#9      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:823:15)
<asynchronous suspension>
#10     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1558:27)
<asynchronous suspension>
#11     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
<asynchronous suspension>
#12     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#13     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:496:9)
<asynchronous suspension>
#14     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
<asynchronous suspension>
#15     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:431:5)
<asynchronous suspension>
#16     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:98:11)
<asynchronous suspension>
#17     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
<asynchronous suspension>
#18     main (package:flutter_tools/executable.dart:99:3)
<asynchronous suspension>

```
</details>

--- a/.github/workflows/windows.yaml
+++ b/.github/workflows/windows.yaml
@@ -63,6 +63,8 @@ jobs:
           npm install -g firebase-tools
       - name: "Build Windows (Release)"
         run: cd tests && flutter build windows --release
+      - name: "Build Windows (Profile)"
+        run: cd tests && flutter build windows --profile
       - name: Start Firebase Emulator and run tests
         run: cd ./.github/workflows/scripts && firebase emulators:exec --project flutterfire-e2e-tests "cd ../../../tests && flutter test .\integration_test\e2e_test.dart -d windows --verbose"
 

--- a/packages/cloud_firestore/cloud_firestore/windows/CMakeLists.txt
+++ b/packages/cloud_firestore/cloud_firestore/windows/CMakeLists.txt
@@ -66,7 +66,12 @@ target_compile_definitions(${PLUGIN_NAME} PRIVATE -DINTERNAL_EXPERIMENTAL=1)
 # Source include directories and library dependencies. Add any plugin-specific
 # dependencies here.
 set(MSVC_RUNTIME_MODE MD)
-set(firebase_libs firebase_core_plugin firebase_auth firebase_firestore)
+# firebase_auth must come after firebase_firestore: the Firestore desktop
+# credentials provider pulls symbols out of firebase_auth (e.g.
+# `firebase::g_auth_initializer`), the Firebase C++ SDK does not declare that
+# dependency between its imported targets, and MSVC resolves static libraries
+# in the order they appear on the link line.
+set(firebase_libs firebase_core_plugin firebase_firestore firebase_auth)
 set(ADDITIONAL_LIBS advapi32 ws2_32 crypt32 rpcrt4 ole32 shell32 Bcrypt.lib DbgHelp.lib)
 target_link_libraries(${PLUGIN_NAME} PRIVATE "${firebase_libs}" "${ADDITIONAL_LIBS}")
 

--- a/packages/cloud_firestore/cloud_firestore/windows/cloud_firestore_plugin.cpp
+++ b/packages/cloud_firestore/cloud_firestore/windows/cloud_firestore_plugin.cpp
@@ -255,7 +255,7 @@ void CloudFirestorePlugin::RegisterWithRegistrar(
 
   auto plugin = std::make_unique<CloudFirestorePlugin>();
 
-  messenger_ = registrar->messenger();
+  plugin->messenger_ = registrar->messenger();
 
   FirebaseFirestoreHostApi::SetUp(registrar->messenger(), plugin.get());
 
@@ -327,9 +327,6 @@ firebase::firestore::FieldValue CloudFirestorePlugin::ConvertToFieldValue(
   }
 }
 
-flutter::BinaryMessenger*
-    cloud_firestore_windows::CloudFirestorePlugin::messenger_ = nullptr;
-
 std::map<std::string,
          std::unique_ptr<flutter::EventChannel<flutter::EncodableValue>>>
     event_channels_;
@@ -343,12 +340,12 @@ std::map<std::string, std::unique_ptr<firebase::firestore::Firestore>>
     cloud_firestore_windows::CloudFirestorePlugin::firestoreInstances_;
 
 std::string RegisterEventChannelWithUUID(
-    std::string prefix, std::string uuid,
+    flutter::BinaryMessenger* messenger, std::string prefix, std::string uuid,
     std::unique_ptr<flutter::StreamHandler<flutter::EncodableValue>> handler) {
   std::string channelName = prefix + uuid;
   event_channels_[channelName] =
       std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
-          CloudFirestorePlugin::messenger_, channelName,
+          messenger, channelName,
           &flutter::StandardMethodCodec::GetInstance(
               &FirebaseFirestoreHostApiCodecSerializer::GetInstance()));
 
@@ -361,7 +358,7 @@ std::string RegisterEventChannelWithUUID(
 }
 
 std::string RegisterEventChannel(
-    std::string prefix,
+    flutter::BinaryMessenger* messenger, std::string prefix,
     std::unique_ptr<flutter::StreamHandler<flutter::EncodableValue>> handler) {
   UUID uuid;
   UuidCreate(&uuid);
@@ -371,7 +368,7 @@ std::string RegisterEventChannel(
   std::string channelName = prefix + str;
   event_channels_[channelName] =
       std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
-          CloudFirestorePlugin::messenger_, channelName,
+          messenger, channelName,
           &flutter::StandardMethodCodec::GetInstance(
               &FirebaseFirestoreHostApiCodecSerializer::GetInstance()));
   stream_handlers_[channelName] = std::move(handler);
@@ -831,7 +828,8 @@ void CloudFirestorePlugin::LoadBundle(
       std::make_unique<LoadBundleStreamHandler>(firestore, bundleConverted);
 
   std::string channelName = RegisterEventChannel(
-      "plugins.flutter.io/firebase_firestore/loadBundle/", std::move(handler));
+      messenger_, "plugins.flutter.io/firebase_firestore/loadBundle/",
+      std::move(handler));
 
   result(channelName);
 }
@@ -1029,7 +1027,7 @@ void CloudFirestorePlugin::SnapshotsInSyncSetup(
   std::string snapshotInSyncId(str);
 
   std::string channelName = RegisterEventChannelWithUUID(
-      "plugins.flutter.io/firebase_firestore/snapshotsInSync/",
+      messenger_, "plugins.flutter.io/firebase_firestore/snapshotsInSync/",
       snapshotInSyncId, std::move(handler));
   result(snapshotInSyncId);
 }
@@ -1202,8 +1200,8 @@ void CloudFirestorePlugin::TransactionCreate(
 
   // Register the event channel.
   std::string channelName = RegisterEventChannelWithUUID(
-      "plugins.flutter.io/firebase_firestore/transaction/", transactionId,
-      std::move(handler));
+      messenger_, "plugins.flutter.io/firebase_firestore/transaction/",
+      transactionId, std::move(handler));
 
   // Return the result (assumed to be transaction ID in this example).
   result(transactionId);
@@ -1864,9 +1862,9 @@ void CloudFirestorePlugin::QuerySnapshot(
       GetServerTimestampBehaviorFromPigeon(
           options.server_timestamp_behavior()));
 
-  std::string channelName =
-      RegisterEventChannel("plugins.flutter.io/firebase_firestore/query/",
-                           std::move(query_snapshot_handler));
+  std::string channelName = RegisterEventChannel(
+      messenger_, "plugins.flutter.io/firebase_firestore/query/",
+      std::move(query_snapshot_handler));
 
   result(channelName);
 }
@@ -1962,9 +1960,9 @@ void CloudFirestorePlugin::DocumentReferenceSnapshot(
           GetServerTimestampBehaviorFromPigeon(
               *parameters.server_timestamp_behavior()));
 
-  std::string channelName =
-      RegisterEventChannel("plugins.flutter.io/firebase_firestore/document/",
-                           std::move(document_snapshot_handler));
+  std::string channelName = RegisterEventChannel(
+      messenger_, "plugins.flutter.io/firebase_firestore/document/",
+      std::move(document_snapshot_handler));
   result(channelName);
 }
 

--- a/packages/cloud_firestore/cloud_firestore/windows/cloud_firestore_plugin.h
+++ b/packages/cloud_firestore/cloud_firestore/windows/cloud_firestore_plugin.h
@@ -176,7 +176,12 @@ class CloudFirestorePlugin : public flutter::Plugin,
       std::function<void(ErrorOr<InternalPipelineSnapshot> reply)> result)
       override;
 
-  static flutter::BinaryMessenger* messenger_;
+  // Per-registration messenger. A Windows app can host several Flutter engines
+  // in the same process (e.g. via `desktop_multi_window`), and each one
+  // registers the plugin with its own registrar. Keeping this per instance
+  // means every channel is built on the messenger of the engine that asked for
+  // it, instead of whichever engine registered last.
+  flutter::BinaryMessenger* messenger_ = nullptr;
   static std::map<
       std::string,
       std::unique_ptr<flutter::EventChannel<flutter::EncodableValue>>>

--- a/packages/firebase_core/firebase_core/windows/CMakeLists.txt
+++ b/packages/firebase_core/firebase_core/windows/CMakeLists.txt
@@ -126,9 +126,17 @@ set(FIREBASE_RELEASE_PATH_LIBS firebase_app firebase_auth firebase_remote_config
 foreach(firebase_lib IN ITEMS ${FIREBASE_RELEASE_PATH_LIBS})
     get_target_property(firebase_lib_path ${firebase_lib} IMPORTED_LOCATION)
     string(REPLACE "Debug" "Release" firebase_lib_release_path ${firebase_lib_path})
+    # Flutter builds Windows apps in Debug, Profile or Release. Profile uses the
+    # release CRT (Flutter maps its compiler/linker flags onto Release), so it
+    # must pick the release libraries too - without an explicit mapping CMake
+    # falls back to plain IMPORTED_LOCATION, which the SDK points at Debug, and
+    # linking fails with unresolved __imp__CrtDbgReport / __imp__invalid_parameter.
     set_target_properties(${firebase_lib} PROPERTIES
       IMPORTED_LOCATION_DEBUG "${firebase_lib_path}"
       IMPORTED_LOCATION_RELEASE "${firebase_lib_release_path}"
+      IMPORTED_LOCATION_PROFILE "${firebase_lib_release_path}"
+      IMPORTED_LOCATION_RELWITHDEBINFO "${firebase_lib_release_path}"
+      IMPORTED_LOCATION_MINSIZEREL "${firebase_lib_release_path}"
     )
 endforeach()
 

